File: app/src/main/java/xyz/lrhm/komakdast/core/util/legacy/ImageKey.java
package xyz.lrhm.komakdast.core.util.legacy;

public class ImageKey {
    String data;

    ImageKey(int resourceId, int width, int height) {
        data = "_" + resourceId + "," + width + "," + height;
    }

    ImageKey(String relativePath, int width, int height) {
        data = "@" + relativePath + "," + width + "," + height;
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ImageKey imageKey = (ImageKey) o;

        if (data != null ? !data.equals(imageKey.data) : imageKey.data != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        return data != null ? data.hashCode() : 0;
    }


    @Override
    public String toString() {
        return data;
    }
}

File: app/src/main/java/xyz/lrhm/komakdast/core/util/legacy/ImageKey.kt
package xyz.lrhm.komakdast.core.util.legacy

class ImageKey {
    var data: String

    internal constructor(resourceId: Int, width: Int, height: Int) {
        data = "_$resourceId,$width,$height"
    }

    internal constructor(relativePath: String, width: Int, height: Int) {
        data = "@$relativePath,$width,$height"
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val imageKey = o as ImageKey
        return if (if (data != null) data != imageKey.data else imageKey.data != null) false else true
    }

    override fun hashCode(): Int {
        return if (data != null) data.hashCode() else 0
    }

    override fun toString(): String {
        return data
    }
}